argusclient/client.py

```diff
@@ -405,8 +407,9 @@ class AlertsServiceClient(BaseUpdatableModelServiceClient):
          Interfaces with the Argus alert notifications endpoint.
 
     """
-    def __init__(self, argus):
-        super(AlertsServiceClient, self).__init__(Alert, argus, "alerts", "alerts/%s")
+    def __init__(self, argus, get_all_path="alerts", get_all_path_params=None):
+        super(AlertsServiceClient, self).__init__(Alert, argus, get_all_path, "alerts/%s",
+                                                  get_all_path_params=get_all_path_params)
```

Comment:
Since this is specific to alerts, we can be more specific here in both naming and assumptions (i.e., assume that the given path is relative to alerts).

```suggestion
    def __init__(self, argus, all_alerts_path=None, all_alerts_params=None):
        get_all_alerts_path = all_alerts_path and f"alerts/{all_alerts_path}" or "alerts"
        super(AlertsServiceClient, self).__init__(Alert, argus, get_all_alerts_path, "alerts/%s",                                                  get_all_params=all_alerts_params)
```

Reply:
Makes sense!

I got the below error because of the f, so I've modified the line
![image](https://user-images.githubusercontent.com/59028215/103928636-2055df00-50ea-11eb-8ad2-47c5f13bcc08.png)

Reply:
and I left it as all_alerts_path to let it be more flexible

Reply:
> I got the below error because of the f, so I've modified the line

You are right, that is a Python 3.0 feature. However, you missed one of my suggestions:

Suggested:

        get_all_alerts_path = all_alerts_path and f"alerts/{all_alerts_path}" or "alerts"

Used:

        get_all_alerts_path = all_alerts_path or "alerts"

Should have been:

        get_all_alerts_path = all_alerts_path and "alerts/" + all_alerts_path or "alerts"

I just want to make sure that your change is intentional. My original intention is to make sure someone doesn't accidentally pass something irrelevant like "namespace/xyz" as path to this constructor.

Reply:
Thanks for making it clear! Yes I actually meant to keep it as I had it, to make it more flexible (as I mentioned above but I guess you missed it), but what you said makes sense too. Will change it